src/MSBuild/TerminalLogger/NodesFrame.cs

```diff
@@ -17,6 +17,7 @@ internal sealed class NodesFrame
     private const int MaxColumn = 120;
 
     private readonly NodeStatus[] _nodes;
+    private readonly int[] _durationLength;
```

Comment:
nit: I am still slightly ruffled by the avoidable allocations made on each frame-render. Would you consider

```suggestion
    private readonly (NodeStatus nodeStatus, int durationLength)[] _nodes;
```

or does it feel like a an unnecessary micro optimization?

Reply:
The allocations were bugging me too, I will absolutely consider that :)